src/components/dailyNotes/NotesList.tsx

```
@@ -7,18 +7,37 @@ import {
+  const isLoading = propNotes ? false : hookData.isLoading;
+  const error = propNotes ? null : hookData.error;
+  const deleteNote = hookData.deleteNote;
+  const updateNote = hookData.updateNote;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [showDeleteConfirm, setShowDeleteConfirm] = useState<string | null>(null);
+  const [showOptions, setShowOptions] = useState<string | null>(null);
+
+  const filteredNotes = notes.filter(note => 
+    note.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
+    note.content.toLowerCase().includes(searchTerm.toLowerCase()) ||
+    (note.tags && note.tags.some(tag => tag.toLowerCase().includes(searchTerm.toLowerCase())))
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)